ui/nuxeo-document-picker/nuxeo-document-picker.js

```diff
@@ -0,0 +1,192 @@
+          }
+          iron-icon:hover {
+            cursor: pointer;
+            opacity: 1;
+          }
+          #mainContainer,
+          #topContainer,
+          nuxeo-results-view {
+            margin: 0;
+            padding: 0;
+          }
+          nuxeo-results-view {
+            --nuxeo-results-view-height: calc(100vh - 520px - var(--nuxeo-app-top));
+          }
+        </style>
+        <nuxeo-dialog id="dialog" modal>
+          <div id="mainContainer">
+            <div id="topContainer">
+              <iron-icon icon="nuxeo:cross" on-tap="_close"></iron-icon>
+            </div>
+            <nuxeo-results-view
+              id="resultsView"
+              provider="[[provider]]"
+              page-size="[[pageSize]]"
+              params="[[params]]"
+              quick-filters="{{quickFilters}}"
+              schemas="[[schemas]]"
+              enrichers="[[enrichers]]"
+              search-name="[[searchName]]"
+              aggregations="{{aggregations}}"
+              visible
+              show-filters
+              opened
+              search-form="[[searchForm]]"
+              on-navigate="_onNavigate"
+              on-results-changed="_onResultsChanged"
+            ></nuxeo-results-view>
+          </div>
+          <div class="buttons">
+            <!-- TODO check if we can have these stuck to the bottom (absolute positioning? check creation wizard) -->
+            <paper-button noink class="secondary" dialog-dismiss>Cancel</paper-button>
+            <paper-button noink class="primary" on-tap="_onInsert" id="insertButton">Insert</paper-button>
+          </div>
+        </nuxeo-dialog>
+      `;
+    }
+
+    static get is() {
+      return 'nuxeo-document-picker';
+    }
+
+    static get properties() {
+      return {
+        aggregations: Object,
+        enrichers: String,
+        pageSize: Number,
+        params: Object,
+        provider: String,
+        quickFilters: Object,
+        schemas: String,
+        searchForm: Object,
+        searchName: String,
+      };
+    }
+
+    ready() {
+      super.ready();
+      if (!this._nxProvider) {
+        this._nxProvider = this.$.provider;
+      }
+    }
+
+    disconnectedCallback() {
+      super.disconnectedCallback();
+      if (this.listenedResults) {
+        this.listenedResults.removeEventListener('selected-items-changed', this._updateInsertButton.bind(this));
+      }
+    }
+
+    open() {
+      if (this.$.resultsView) {
+        // XXX could make sense to consider having this as a public method on nuxeo-results-view
+        this.$.resultsView._clear();
+      }
+      this._updateInsertButton();
+      this.$.dialog.open();
+    }
+
+    _close() {
+      this.$.dialog.close();
+    }
+
+    _getSelectedItems() {
+      return (
+        this.$.resultsView &&
+        this.$.resultsView.results &&
+        this.$.resultsView.results.results &&
+        this.$.resultsView.results.results.selectedItems
+      );
```

Comment:
We could use conditional chaining here, right? Something like:
```suggestion
        return this.$.resultsView?.results?.results?.selectedItems;
```

Reply:
I'm a big fan of this, I didn't take it because I don't think I've seen it used on Elements or Web UI before.
([we should be safe in terms of browser compat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining#browser_compatibility))

Reply:
It seems that it break linting:
```
ui/nuxeo-document-picker/nuxeo-document-picker.js(NaN,NaN) error [parse-error] - This experimental syntax requires enabling the parser plugin: 'optionalChaining' (172:31)
```

Reply:
If we don't have a good way to solve this, we could create a ticket to clean it up afterwards.

Reply:
ELEMENTS-1320: Allow optional chaining in Nuxeo Elements